pkg/exemplar/exemplar.go

```diff
@@ -13,12 +13,86 @@
 
 package exemplar
 
-import "github.com/prometheus/prometheus/pkg/labels"
+import (
+	"bytes"
+	"encoding/json"
+	"math"
+	"strconv"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+)
 
 // Exemplar is additional information associated with a time series.
 type Exemplar struct {
 	Labels labels.Labels
 	Value  float64
-	HasTs  bool
 	Ts     int64
+	HasTs  bool
```

Comment:
nit: is this needed? lets not reorder the fields.

Reply:
Exemplars can not-have their own timestamp, so this is needed so that we know whether to take this `Ts` or the one from the sample.

Reply:
Or this could rather be something returned by the parser interface.

Reply:
(to be clear: the comment is about re-order the field in the struct, not about the field itself)

Reply:
But turns out we don't need that field at all and can enforce attaching "some" timestamp.

Reply:
This is used to deduplicate exemplars that were exposed without timestamp, but we could find another way to check that.